### PR TITLE
Fix: Hero image in light mode

### PR DESCRIPTION
**Pull request by ahqsoftwares**
#### What kind of changes does this PR include?
- Minor content fixes (broken links, typos, etc.)

#### Description
- What does this PR change? Give us a brief description.
> It's just a fix for the broken hero icon in the light mode
- Did you change something visual? A before/after screenshot can be helpful.
> Yes, the animation has been changed a little bit

https://github.com/tauri-apps/tauri-docs/assets/84524025/8720643b-ce3e-4db7-aa4b-4a28ed9a1957

### CODE REVIEW

**Review by lorenzolewis — CHANGES_REQUESTED** [review]
Thanks for this @ahqsoftwares ! This solves the problem perfectly for the different colour modes 🎨 Two bits of feedback

- https://github.com/tauri-apps/tauri-docs/pull/1511 was just merged in, would you mind adding in the respective change for the French page as well alongside this?
- I did a few tweaks on the CSS to simplify it and let the parent elements take care of the spacing instead of hard-coding it. Hope you don't mind the commit on your PR.
- I'm personally not a fan of infinite CSS animations as they can be distracting to the reader and prefer the "settle".

Let me know if you'd be able to make that change on the French page (you may need to merge from `next` into your branch for that French translation to show up) and if you're okay with the CSS changes. Thank you!

**Comment by lorenzolewis** [comment]
Ended up breaking the build when I merged it, so went ahead and fixed the French index page 😅 

Let me know if this all looks okay and sorry for high-jacking this PR

**Review by lorenzolewis — APPROVED** [review]

**Comment by simonhyll** [comment]
I would like to change it to a component instead of adding the entire SVG into the index.mdx file

**Comment by simonhyll** [comment]
As a component you can instead import the svg directly so updating the hero is as easy as replacing the svg file

**Comment by ahqsoftwares** [comment]
> I would like to change it to a component instead of adding the entire SVG into the index.mdx file

If i import svg, directly, I lose every way to  use css vars as the strokes, the best is using astro Component

**Comment by ahqsoftwares** [comment]
@simonhyll Implementing the changes you requested

**Comment by ahqsoftwares** [comment]
> Thanks for this @ahqsoftwares ! This solves the problem perfectly for the different colour modes 🎨 Two bits of feedback
> 
> * [i18n(fr): Translate `index.mdx`, `frontend/index.mdx`, `nextjs.mdx`, `nuxt.mdx`, `trunk.mdx` #1511](https://github.com/tauri-apps/tauri-docs/pull/1511) was just merged in, would you mind adding in the respective change for the French page as well alongside this?
> * I did a few tweaks on the CSS to simplify it and let the parent elements take care of the spacing instead of hard-coding it. Hope you don't mind the commit on your PR.
> * I'm personally not a fan of infinite CSS animations as they can be distracting to the reader and prefer the "settle".
> 
> Let me know if you'd be able to make that change on the French page (you may need to merge from `next` into your branch for that French translation to show up) and if you're okay with the CSS changes. Thank you!

Yeah no problem

**Comment by ahqsoftwares** [comment]
@simonhyll Using Astro Component is increasing Cumulative Layout Shift

**Comment by simonhyll** [comment]
@ahqsoftwares If done right there should be zero difference between what you did initially and using a component. The dom renders the same

**Comment by ahqsoftwares** [comment]
> @ahqsoftwares If done right there should be zero difference between what you did initially and using a component. The dom renders the same

Can you please take a look at the current commit

**Comment by simonhyll** [comment]
@ahqsoftwares Should be shifting because you do it at runtime instead of at compile time. Check the IPC blog post to see how I load it at compile time instead. 
<https://github.com/tauri-apps/tauri-docs/blob/4774cc6869c45e0e831ad59ea08d16398ab8855a/src/content/docs/blog/2023-08-24.mdx#L6-L8>

**Comment by lorenzolewis** [comment]
@ahqsoftwares not sure if you saw this message on Discord, if should be an easy fix that's super minimal for all of this: https://discord.com/channels/616186924390023171/662624589037436928/1149951658474864641

```css
html[data-theme='light'] .hero > img {
    filter: invert(1);
}
```

You can use the above with the way everything is set up currently on the `next` branch without having to go into JS territory

**Comment by ahqsoftwares** [comment]
> @ahqsoftwares not sure if you saw this message on Discord, if should be an easy fix that's super minimal for all of this: https://discord.com/channels/616186924390023171/662624589037436928/1149951658474864641
> 
> ```css
> html[data-theme='light'] .hero > img {
>     filter: invert(1);
> }
> ```
> 
> You can use the above with the way everything is set up currently on the `next` branch without having to go into JS territory

💀

**Comment by ahqsoftwares** [comment]
Im make a clean start

**Comment by lorenzolewis** [comment]
Hehe sorry, the land of SVGs always unlock all the complexity 😅

**Comment by ahqsoftwares** [comment]
Still need to confirm if data-theme="light" is loaded by astro or not. Is that css snippet tested?

**Comment by ahqsoftwares** [comment]
image
HTML is so weird, everythie the impl is different, I was thinking like there would be a tailwind like dark class

**Comment by ahqsoftwares** [comment]
@lorenzolewis Everything's done now

**Comment by lorenzolewis** [comment]
It's Tailwind-inspired, Tailwind actually let's you do a similar approach if you go in and adjust the config.

I've seen a bit of research on the differences of performance in using `class` vs. `id` vs. `data-*` vs. element selectors like `div` and it's a strange, strange land. I'm not 100% sure if all of this is driven off of that, but I think Astro is moving towards `data-*` style selectors if memory serves correctly 😅

**Comment by ahqsoftwares** [comment]
HTML is kinda opinionated, even you don't have the same type of markup / styling on all projects

**Comment by ahqsoftwares** [comment]
@lorenzolewis You may test it now

File: src/styles/custom.css [hunk 1]
```diff
@@ -122,13 +117,29 @@
...
+		scale: 1.5;
+		opacity: 0;
+	}
+
+	80% {
+		scale: 0.9;
 	}
 
 	100% {
+		scake: 1.4;
```
Comment (lorenzolewis) [inline src/styles/custom.css]:
```suggestion
		scale: 1.4;
```

I need to test what this looks like on mobile, but just fixing a quick typo first ;)

Comment (ahqsoftwares) [reply src/styles/custom.css]: Well, now I know why I shouldn't be coding at 1AM midnight lol

**Comment by ahqsoftwares** [comment]
@lorenzolewis I think for mobile, no animations, no transition and making it small would be good?

**Comment by lorenzolewis** [comment]
Let me check with the latest commit you just pushed and see what it looks like after it builds. The change you did is exactly what I was going to call out because it looked a little funky on smaller viewports, but let's see what this looks like after it builds.

**Comment by ahqsoftwares** [comment]
I might be using starlight for my docs page too, it's kinda easy

File: src/styles/custom.css [hunk 2]
```diff
@@ -122,14 +117,28 @@
 	animation: 14s intro;
 }
 
+/* Tauri logo on index page */
+.hero > img {
+	animation: 3s intro;
+	opacity: 0.4;
+	width: 100%;
+}
```
Comment (lorenzolewis) [inline src/styles/custom.css]:
```suggestion
	animation: 3s intro;
	opacity: 0.4;
}
```

This was left over from the previous impl I believe and making it take up more screen space than needed on mobile.
Before: 
Screenshot 2023-09-09 at 08 11 47

After:
Screenshot 2023-09-09 at 08 12 09

**Comment by lorenzolewis** [comment]
> I might be using starlight for my docs page too, it's kinda easy

It's awesome, and the Astro community is amazing

File: src/styles/custom.css (hunk 2 above)
Comment (ahqsoftwares) [reply src/styles/custom.css]: I had made the fix already but forgot to push it. Currently in a power ot, I shall fix when it comes (~30min)

**Review by lorenzolewis — APPROVED** [review]
Looks awesome, and thank you for going along with @simonhyll and myself on our adventure through the joy of CSS, SVG, Astro components, and HTML on this one @ahqsoftwares 😅

**Comment by ahqsoftwares** [comment]
# 😅

**Pull request merged**